public/app/features/trails/TrailStore/TrailStore.ts

```diff
@@ -132,3 +142,26 @@ export function getTrailStore(): TrailStore {
 
   return store;
 }
+
+function getCurrentUrlValues({ history, currentStep }: SerializedTrail) {
+  return history[currentStep]?.urlValues || history.at(-1)?.urlValues;
+}
+
+function relevantUrlValuesAreEquivalent(a: SceneObjectUrlValues, b: SceneObjectUrlValues) {
+  const baseValuesMatch =
+    a.metric === b.metric &&
+    a.from === b.from &&
+    a.to === b.to &&
+    a['var-ds'] === b['var-ds'] &&
+    a['var-groupby'] === b['var-groupby'];
+
+  const [aFilters, bFilters] = [a, b].map((vals) => {
+    // ensure type is for each filter string[]
+    const filters = typeof vals['var-filters'] === 'string' ? [vals['var-filters']] : vals['var-filters'] || [];
+    return filters;
+  });
+  const filtersMatch =
+    aFilters.every((filter) => bFilters.includes(filter)) && bFilters.every((filter) => aFilters.includes(filter));
+
+  return baseValuesMatch && filtersMatch;
+}
```

Comment:
This seems so overly complex for something that should be just a single-string comparison (of the complete url state). if the trail store just stored the url state string (and used this when opening bookmarks), could we not just compare that?

Reply:
or maybe lodash isEqual could here?

Reply:
would be nice to use soemthing that is agnostic to the actual url state so we do not need to modify this if we add more to url state